cmd/swapd/main.go

```diff
@@ -470,8 +457,9 @@ func newBackend(
 
 	contractAddrStr := c.String(flagContractAddress)
 	if contractAddrStr != "" {
-		// We check the contract code at the address later, so we don't need
-		// to tightly validate the address here.
```

Comment:
I added that comment in a previous PR, as it was bothering me that were not error checking. The comment was incorrect.  On error we would get an address of all zeros and would not validate the contract code.